Only in gd-2.0.35.new/: .configure.ac.swp
diff -ur gd-2.0.35/Makefile.am gd-2.0.35.new/Makefile.am
--- gd-2.0.35/Makefile.am	2007-03-05 16:42:16.000000000 +0100
+++ gd-2.0.35.new/Makefile.am	2010-01-14 01:23:57.218750000 +0100
@@ -19,9 +19,9 @@
 
 libgd_la_SOURCES = gd.c gdfx.c gd_security.c gd_gd.c gd_gd2.c gd_io.c gd_io_dp.c gd_gif_in.c gd_gif_out.c gd_io_file.c gd_io_ss.c gd_jpeg.c gd_png.c gd_ss.c gd_topal.c gd_wbmp.c gdcache.c gdfontg.c gdfontl.c gdfontmb.c gdfonts.c gdfontt.c gdft.c gdhelpers.c gdhelpers.h gdkanji.c gdtables.c gdxpm.c jisx0208.h wbmp.c wbmp.h
 
-libgd_la_LDFLAGS = -version-info 2:0:0 $(XTRA_LDFLAGS)
+libgd_la_LDFLAGS = -version-info 2:0:0 -no-undefined $(XTRA_LDFLAGS) -no-undefined 
 
-libgd_la_LIBADD = $(LTLIBICONV)
+libgd_la_LIBADD = $(LTLIBICONV) -lpng
 
 LDADD = ./libgd.la $(LIBICONV)
 
diff -ur gd-2.0.35/Makefile.in gd-2.0.35.new/Makefile.in
--- gd-2.0.35/Makefile.in	2007-04-23 16:57:51.000000000 +0200
+++ gd-2.0.35.new/Makefile.in	2010-01-14 01:34:06.765625000 +0100
@@ -183,7 +183,7 @@
 	$(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+	$(AM_LDFLAGS) $(LDFLAGS) -o $@ -no-undefined
 SOURCES = $(libgd_la_SOURCES) annotate.c circletexttest.c \
 	fontconfigtest.c fontsizetest.c fontwheeltest.c gd2copypal.c \
 	gd2time.c gd2togif.c gd2topng.c gdcmpgif.c gddemo.c \
@@ -345,9 +345,9 @@
 include_HEADERS = gd.h gdfx.h gd_io.h gdcache.h gdfontg.h gdfontl.h gdfontmb.h gdfonts.h gdfontt.h entities.h
 lib_LTLIBRARIES = libgd.la
 libgd_la_SOURCES = gd.c gdfx.c gd_security.c gd_gd.c gd_gd2.c gd_io.c gd_io_dp.c gd_gif_in.c gd_gif_out.c gd_io_file.c gd_io_ss.c gd_jpeg.c gd_png.c gd_ss.c gd_topal.c gd_wbmp.c gdcache.c gdfontg.c gdfontl.c gdfontmb.c gdfonts.c gdfontt.c gdft.c gdhelpers.c gdhelpers.h gdkanji.c gdtables.c gdxpm.c jisx0208.h wbmp.c wbmp.h
-libgd_la_LDFLAGS = -version-info 2:0:0 $(XTRA_LDFLAGS)
-libgd_la_LIBADD = $(LTLIBICONV)
-LDADD = ./libgd.la $(LIBICONV)
+libgd_la_LDFLAGS = -version-info 2:0:0 $(XTRA_LDFLAGS) -no-undefined
+libgd_la_LIBADD = $(LTLIBICONV) -no-undefined -lpng
+LDADD = ./libgd.la $(LIBICONV) -lpng
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
@@ -431,7 +431,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libgd.la: $(libgd_la_OBJECTS) $(libgd_la_DEPENDENCIES) 
-	$(LINK) -rpath $(libdir) $(libgd_la_LDFLAGS) $(libgd_la_OBJECTS) $(libgd_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(libdir) $(libgd_la_LDFLAGS) $(libgd_la_OBJECTS) $(libgd_la_LIBADD) $(LIBS) -lpng -no-undefined
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(mkdir_p) "$(DESTDIR)$(bindir)"
diff -ur gd-2.0.35/configure.ac gd-2.0.35.new/configure.ac
--- gd-2.0.35/configure.ac	2007-04-25 14:48:41.000000000 +0200
+++ gd-2.0.35.new/configure.ac	2010-01-14 10:59:42.234375000 +0100
@@ -407,6 +407,10 @@
 if test "$ac_cv_lib_png12_png_create_read_struct" = yes; then
   ac_cv_lib_png_png_create_read_struct=yes
 fi 
+if test "$ac_cv_lib_png_3_png_create_read_struct" = yes; then
+  ac_cv_lib_png_png_create_read_struct=yes
+fi 
+
 
 dnl FreeType configure tests snarfed from libwmf ..
 AC_ARG_WITH(freetype,dnl
